src/server/container_utils.cc

```diff
@@ -23,6 +23,89 @@ extern "C" {
 
 namespace dfly::container_utils {
 
+namespace {
+
+struct ShardFFResult {
+  PrimeKey key;
+  ShardId sid = kInvalidSid;
+};
+
+OpResult<std::string> FindFirstNonEmptySingleShard(Transaction* trans, int req_obj_type,
```

Comment:
how can I understand what the differences are? :)

Reply:
1. The caller site has a comment what the fast and slow paths are and why they are so
2. Those functions didn't have comments before
3. Added brief comments to them